### ISSUE TITLE
Trying to clone a local repository using a file:// URI doesn't work on Windows

### ISSUE THREAD

**Issue by silvanocerza** [go-git/go-git#247]
If I try to clone a local repository on Windows using an URI similar to `file:///C:/Users/My/Cool/Project/` it fails with `Error: repository not found`. The same URI works without issues when calling the good old Git from command line. Using a Windows path like `C:\\Users\\My\\Cool\\Project\\` works with `go-git` and Git as expected. 

This happens only on Windows, Linux and OS X work nicely.

Git urls docs for reference: https://www.git-scm.com/docs/git-clone#_git_urls

I've made a minimal project to reproduce the issue: https://github.com/silvanocerza/go-git-file-uri-issue

**Comment by lindell** [comment]
I had a similar problem trying to run tests on [this project](https://github.com/lindell/multi-gitter), but have found out some things around go-git and local filesystem-cloning during some research. 

First of all, this change should is working **most of the time** on Windows.

```go
firstRepoFileURI := fmt.Sprintf("file:///%s", filepath.ToSlash(firstRepoLocalPath))
```
do this instead⬇️ 
```go
firstRepoFileURI := fmt.Sprintf(`file://"%s"`, filepath.ToSlash(firstRepoLocalPath))
```

It has been working on my local machine. But I have struggled to get it to work on GitHub actions. My assumption before was that there was a difference between Windows 10 and Windows Server.

But after forking @silvanocerza's example project and making some changes that passed what in my project failed, I found the root cause.
Added GitHub actions with @silvanocerza's code (**failing test**): https://github.com/lindell/go-git-file-uri-issue/commit/6c5ead56308c4c4ad45efabb67f75372014bf695
Making the suggested change (**passing test**): https://github.com/lindell/go-git-file-uri-issue/commit/ed8e7d87bfc08ad8a0805d6d8b82a118e55f1cf5
Using a temporary directory (**failing test**): https://github.com/lindell/go-git-file-uri-issue/commit/23b9f3cd70b7a999e19df8ded3a019e47eb41c8e

**The problem seems to be if the path is on another drive than the working directory!**

I can now reproduce this bug on my local machine in the multi-gitter project by running the tests on another drive.
